klt-cert-manager/controllers/keptnwebhookcontroller/certs.go

```diff
@@ -0,0 +1,226 @@
+package keptnwebhookcontroller
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/keptn/lifecycle-toolkit/klt-cert-manager/kubeutils"
+)
+
+var serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), 128)
+
+const (
+	renewalThreshold = 12 * time.Hour
+
+	RootKey     = "ca.key"
+	RootCert    = "ca.crt"
+	RootCertOld = "ca.crt.old"
+	ServerKey   = "tls.key"
+	ServerCert  = "tls.crt"
+)
+
+// Certs handles creation and renewal of CA and SSL/TLS server certificates.
+type Certs struct {
+	Domain  string
+	SrcData map[string][]byte
+	Data    map[string][]byte
+
+	Now time.Time
+
+	rootPrivateKey *ecdsa.PrivateKey
+	rootPublicCert *x509.Certificate
+}
+
+// ValidateCerts checks for certificates and keys on cs.SrcData and renews them if needed. The existing (or new)
+// certificates will be stored on cs.Data.
+func (cs *Certs) ValidateCerts() error {
+	cs.Data = map[string][]byte{}
+	if cs.SrcData != nil {
+		for k, v := range cs.SrcData {
+			cs.Data[k] = v
+		}
+	}
+
+	now := time.Now().UTC()
+	if !cs.Now.IsZero() {
+		now = cs.Now
+	}
+
+	renewRootCerts := cs.validateRootCerts(now)
+	if renewRootCerts {
+		if err := cs.generateRootCerts(cs.Domain, now); err != nil {
+			return err
+		}
+	}
+
+	if renewRootCerts || cs.validateServerCerts(now) {
+		return cs.generateServerCerts(cs.Domain, now)
+	}
+
+	return nil
+}
+
+func (cs *Certs) validateRootCerts(now time.Time) bool {
+	if cs.Data[RootKey] == nil || cs.Data[RootCert] == nil {
+		////log.Info("no root certificates found, creating")
```

Comment:
do we want to keep the logs in this file? If yes pls uncomment, if no, pls delete